### DISCUSSION TITLE
Web3Modal V3.1  switchNetwork/defaultChains in React/wagmi.

### DISCUSSION THREAD

**Discussion by ineskh** [WalletConnect/walletconnect-monorepo#3787] [Developer Support] [answered]
Hello,

I'm currently using the following documentation to integrate Web3Modal into my Dapp: [Web3Modal Integration Guide](https://docs.walletconnect.com/web3modal/v2/javascript/wagmi/custom-chains).

I have a question: How can I dynamically set or switch to a different blockchain (with wagmi) without relying on the modal to predefine it? For example, if my component calls this function:

createWeb3Modal({ wagmiConfig, projectId, defaultChain: mainnet, chains: [mainnet, polygon] })

And then, if the user decides to change to the Polygon network using my Dapp, the component will render with:

createWeb3Modal({ wagmiConfig, projectId, defaultChain: polygon, chains: [mainnet, polygon] })

However, when I check the state, selectedNetworkId is set to 1, whereas I expect it to be selectedNetworkId: 137 (Polygon). How can I ensure that the correct network is selected dynamically?

Thank you so much for your help.

**Comment by glitch-txs** [comment, marked as answer]
Hi, this is not possible at the moment, however you can open an issue in Web3Modal repository with this feature request
